### DISCUSSION TITLE
Creating additional libs in packages

### DISCUSSION THREAD

**Discussion by regenrek** [cawa-93/vite-electron-builder#636] [Q&A] [answered]
Hi,

any ideas how I can achieve to create a local lib like in a monorepo setup?

```
packages/
  core/
  -- dist/
  -- src/
  --- index.ts
  main/
  preloader/
  renderer/
```

renderer/src/index.ts
```
// imports...
import store from '/@/store';
import environment from '../../core/src/index';

createApp(App)
  .use(router)
  .use(store)
  .mount('#app');

environment.init(store);
```

I run always into some errors.

![image](https://user-images.githubusercontent.com/5182020/147400963-da6ab588-81fd-44d2-bfd1-5620e1855a9e.png)

![image](https://user-images.githubusercontent.com/5182020/147400964-d6cdcbd6-f43e-4dae-8c24-7b78cb0a45fe.png)

**Comment by cawa-93** [comment, marked as answer]
`__dirname` is part of Node.js CJS modules API and not awailable in Renderer process.

**Comment by regenrek** [comment]
My mistake. I removed the vite.config now from the shared package. This repo has a shared package which is kind what I've wanted hto achieve. Thanks! ttps://github.com/cawa-93/anime-library
